pkg/policy/selectorcache.go

```diff
@@ -597,11 +597,15 @@ func (sc *SelectorCache) UpdateIdentities(added, deleted cache.IdentityCache, wg
 				}
 			}
 			for numericID := range added {
-				if _, exists := idSel.cachedSelections[numericID]; !exists {
-					if idSel.source.matches(sc.idCache[numericID]) {
-						adds = append(adds, numericID)
-						idSel.cachedSelections[numericID] = struct{}{}
-					}
+				matches := idSel.source.matches(sc.idCache[numericID])
+				_, exists := idSel.cachedSelections[numericID]
```

Comment:
Previously, the `matches` called occurred if the ID didn't already exist, but now `matches` is always called. Is this significant?

Reply:
Right, that behavior was wrong (and I wrote it :upside_down_face:) because it didn't correctly handle the case where an identity was mutated.

We can't cache this; we need to check every time. This isn't so bad, though, because in the code above, we already elide all identities that aren't net-new or mutating. So, this shouldn't actually add to many new comparisons.

Reply:
And when we say mutating identities, we are only talking about the special case of `reserved:host` with `kube-apiserver`, right?

Reply:
exactly right, identity 1, a.k.a `reserved:host`, can change its set of labels. For all other identities the set of labels *should* be immutable, unless we somehow miss a delete + add.